### PR TITLE
Specify generated assets style

### PR DESCRIPTION
**Pull request by lcdsmao**
### What does this change?
Add support to generate flat hierarchy assets with field name style:
- camel-case
- snake-case

### What is the value of this and can you measure success?
CI pass

### CODE REVIEW

File: lib/src/generators/assets_generator.dart [hunk 1]
```diff
@@ -3,66 +3,53 @@ import 'dart:io';
...
-        }
-      }
+  String assetsGenStyle;
+  if (flutterGen != null && flutterGen.hasAssets) {
+    assetsGenStyle = flutterGen.assets.style;
+  }
 
-      assetTypeQueue.addAll(assetType.children);
-    }
+  if (assetsGenStyle == 'dot-delimiter' || assetsGenStyle == null) {
```
Comment (wasabeef) [inline lib/src/generators/assets_generator.dart:40]: How do you think? 🤔

```dart
  if (flutterGen == null ||
      !flutterGen.hasAssets ||
      flutterGen.assets.isDefault) {
    classesBuffer.writeln(
        _dotDelimiterStyleDefinition(pubspecFile, assets, integrations));
  } else if (flutterGen.assets.isSnakeCase) {
    classesBuffer
        .writeln(_snakeCaseStyleDefinition(pubspecFile, assets, integrations));
  } else if (flutterGen.assets.isCamelCase) {
    classesBuffer
        .writeln(_camelCaseStyleDefinition(pubspecFile, assets, integrations));
  } else {
    throw 'The value of "flutter_gen/assets/style." is incorrect.';
  }
```

```dart
class FlutterGenAssets {
  // ...
  bool get isDefault => style == null || isDotDelimiter;
  bool get isDotDelimiter => style == 'dot-delimiter';
  bool get isSnakeCase => style == 'snake-case';
  bool get isCamelCase => style == 'camel-case';
}
```

File: lib/src/generators/assets_generator.dart [hunk 2]
```diff
@@ -3,66 +3,53 @@ import 'dart:io';
...
+    classesBuffer.writeln(
+        _dotDelimiterStyleDefinition(pubspecFile, assets, integrations));
+  } else if (assetsGenStyle == 'snake-case') {
+    classesBuffer
+        .writeln(_snakeCaseStyleDefinition(pubspecFile, assets, integrations));
+  } else if (assetsGenStyle == 'camel-case') {
+    classesBuffer
+        .writeln(_camelCaseStyleDefinition(pubspecFile, assets, integrations));
+  } else {
+    throw 'The value of "flutter_gen/assets/style." is incorrect.';
```
Comment (wasabeef) [inline lib/src/generators/assets_generator.dart:50]: And.

It doesn't work on my mac when style is camel-case or snake-case.
```
FlutterGen Loading ... example/pubspec.yaml
Unhandled exception:
NoSuchMethodError: The method 'toStaticFieldString' was called on null.
Receiver: null
Tried calling: toStaticFieldString()
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      _assetsClassDefinition.<anonymous closure> (package:flutter_gen/src/generators/assets_generator.dart:275:31)
#2      MappedListIterable.elementAt (dart:_internal/iterable.dart:417:31)
#3      ListIterable.join (dart:_internal/iterable.dart:149:25)
#4      _assetsClassDefinition (package:flutter_gen/src/generators/assets_generator.dart:277:8)
#5      _flatStyleDefinition (package:flutter_gen/src/generators/assets_generator.dart:269:10)
#6      _camelCaseStyleDefinition (package:flutter_gen/src/generators/assets_generator.dart:225:10)
#7      generateAssets (package:flutter_gen/src/generators/assets_generator.dart:48:18)
#8      FlutterGenerator.build (package:flutter_gen/flutter_generator.dart:67:27)
<asynchronous suspension>
#9      main (file:///Users/a12622/git/flutter_gen/bin/flutter_gen_command.dart:37:48)
#10     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#11     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
```

Comment (lcdsmao) [reply lib/src/generators/assets_generator.dart:47]: Fixed!

**Review by wasabeef — APPROVED** [review]
Good Good Good.

Well, work!!

**Pull request merged**